collector/zfs_linux.go

```diff
@@ -81,6 +98,50 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 	return scanner.Err()
 }
 
-func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+func (c *zfsCollector) parsePoolProcfsFiles(zpoolPaths []string, handler func(string, zfsSysctl, int)) (err error) {
+
+	for _, zpoolPath := range zpoolPaths {
+		file, err := os.Open(zpoolPath)
+		if err != nil {
+			log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", zpoolPath)
+			err = zfsNotAvailableError
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+
+		parseLine := false
+		var fields []string
+		for scanner.Scan() {
+
+			line := strings.Fields(scanner.Text())
+
+			if !parseLine && len(line) >= 12 && line[0] == "nread" {
+				//Start parsing from here.
+				parseLine = true
+				fields = make([]string, len(line))
+				copy(fields, line)
+				continue
+			}
+			if !parseLine {
+				continue
+			}
+
+			zpoolPathElements := strings.Split(zpoolPath, "/")
```

Comment:
I covered the rest of your suggestions, but I wasn't quite sure what you were asking for here. Do you want me to have a hard-coded known value for the maximum set of values that I expect? I hesitate to do that...for example, the length is different right now between the e2e-tests faked out path and the actual expected runtime proc path. I could still update to make sure the count falls in that expected range, I guess. Let me know if I misunderstood.